crates/key-value-azure/src/store.rs

```diff
@@ -154,6 +154,15 @@ impl Store for AzureCosmosStore {
     }
 
     async fn set(&self, key: &str, value: &[u8]) -> Result<(), Error> {
+        let illegal_chars = ['/', '\\', '?', '#'];
+
+        if key.contains(|c| illegal_chars.contains(&c)) {
+            return Err(Error::Other(format!(
+                "Key contains an illegal character. Allowed characters do not include: {}",
```

Comment:
The wording here reads a bit awkwardly to me.  Would you be open to e.g.

```suggestion
                "Key contains an illegal character. Keys must not include any of: {}",
```

or

```
"Key contains an illegal character. The illegal characters are: {}",
```